src/hotspot/share/ci/ciEnv.cpp

```diff
@@ -516,16 +516,10 @@ ciKlass* ciEnv::get_klass_by_name_impl(ciKlass* accessing_klass,
   }
 
   Klass* found_klass;
-  {
-    ttyUnlocker ttyul;  // release tty lock to avoid ordering problems
-    MutexLocker ml(current, Compile_lock);
-    Klass* kls;
-    if (!require_local) {
-      kls = SystemDictionary::find_constrained_instance_or_array_klass(current, sym, loader);
-    } else {
-      kls = SystemDictionary::find_instance_or_array_klass(current, sym, loader, domain);
-    }
-    found_klass = kls;
+  if (!require_local) {
```

Comment:
Since JVMCI needs the very same code, does it make sense to put it into a wrapper method?

Reply:
This is a good idea.